src/main/java/uk/gov/hmcts/sptribs/common/event/page/PartiesToContact.java

```diff
@@ -27,28 +28,39 @@ public void addTo(PageBuilder pageBuilder) {
             .pageLabel("Contact Parties")
             .label("LabelPartiesToContactMessage", "Which parties do you want to contact?")
             .complex(CaseData::getCicCase)
-            .optionalWithLabel(CicCase::getNotifyPartySubject, RECIPIENT_LABEL)
+            .readonly(CicCase::getFullName, ALWAYS_HIDE)
+            .optionalWithoutDefaultValue(CicCase::getNotifyPartySubject,
+                "cicCaseFullName!=\"\" ", RECIPIENT_LABEL)
             .readonly(CicCase::getApplicantFullName, ALWAYS_HIDE)
             .optionalWithoutDefaultValue(CicCase::getNotifyPartyApplicant,
                 "cicCaseApplicantFullName!=\"\" ", RECIPIENT_LABEL)
             .readonly(CicCase::getRepresentativeFullName, ALWAYS_HIDE)
             .optionalWithoutDefaultValue(CicCase::getNotifyPartyRepresentative,
                 "cicCaseRepresentativeFullName!=\"\" ", RECIPIENT_LABEL)
-            .optionalWithLabel(CicCase::getNotifyPartyRespondent, RECIPIENT_LABEL)
-
+            .readonly(CicCase::getRespondentName, ALWAYS_HIDE)
+            .optionalWithoutDefaultValue(CicCase::getNotifyPartyRespondent,
+                "cicCaseRespondentName!=\"\" ", RECIPIENT_LABEL)
+            .mandatory(CicCase::getNotifyPartyMessage)
             .done();
     }
 
     public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseData, State> details,
                                                                   CaseDetails<CaseData, State> detailsBefore) {
         final CaseData data = details.getData();
-        final List<String> errors = EventUtil.checkRecipient(data);
+        final List<String> errors = new ArrayList<>();
+
+
+        if (null != data.getContactParties() && CollectionUtils.isEmpty(data.getCicCase().getNotifyPartySubject())
```

Comment:
A minor comment - we can move data.getCicCase() to a variable due to the repeated usage in the if condition.